semantic-conventions/src/opentelemetry/semconv/model/semantic_convention.py

```diff
@@ -431,32 +431,15 @@ def _populate_extends_single(self, semconv, unprocessed):
             parent_attributes = {}
             for ext_attr in extended.attributes:
                 parent_attributes[ext_attr.fqn] = ext_attr.inherit_attribute()
-            # By induction, parent semconv is already correctly sorted
-            parent_attributes.update(
-                SemanticConventionSet._sort_attributes_dict(semconv.attrs_by_name)
-            )
-            if parent_attributes or semconv.attributes:
-                semconv.attrs_by_name = parent_attributes
-        elif semconv.attributes:  # No parent, sort of current attributes
-            semconv.attrs_by_name = SemanticConventionSet._sort_attributes_dict(
-                semconv.attrs_by_name
-            )
+
+            parent_attributes.update(semconv.attrs_by_name)
+            semconv.attrs_by_name = parent_attributes
+
+        semconv.attrs_by_name = dict(sorted((semconv.attrs_by_name.items())))
+
```

Comment:
I just tested this. It seems that the sorting is not being applied on attribute references.

In particular, the test in `tests/data/markdown/ref` still passes successfully with the following expected output (which is not being sorted alphabetically):

```
| Attribute  | Type | Description  | Examples  | Requirement Level |
|---|---|---|---|---|
| `rpc.service` | string | The service name, must be equal to the $service part in the span name. | `EchoService` | Required |
| [`net.peer.name`](input_general.md) | string | override brief. [1] | `example.com` | Opt-In |
...
```
Here, `net.*` attributes are listed _after_ the `rpc.service` attribute just because those are references.

- I think, we need to adapt that test with the expected sorted output and **fix the sorting to work for references** as well
- An explicit test (or adapting the existent test) for making sure the sorting also works for template-type attributes would be good as well

Reply:
Thanks a lot @AlexanderWert! I believe I fixed it.